cmake/TorchConfig.cmake.in

```diff
@@ -166,9 +166,11 @@ if(@USE_CUDA@)
   list(APPEND TORCH_LIBRARIES ${TORCH_CUDA_LIBRARIES})
 endif()
 
-# When we build libtorch with the old GCC ABI, dependent libraries must too.
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+# When we build libtorch with the old libstdc++ ABI, dependent libraries must too.
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set(TORCH_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=@GLIBCXX_USE_CXX11_ABI@")
+else()
+  set(TORCH_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=0")
```

Comment:
Can you please elaborate why this is needed? This define should not do anything on non-Linux platforms, should it?

Reply:
It shouldn't be needed. Just trying to figure out why most of the builds have started to fail after this change with a `error: invalid token at start of a preprocessor expression`.

Reply:
You probably just need to rebase your PR against commit from `viable/strict` (there were several regressions this morning)

Reply:
Ok, resolved the issue. Apparently caffe2's CMakeLists.txt was the culprit and had a similar check that caused GLIBCXX_USE_CXX11_ABI  variable to become empty. Everything is green now.